### ISSUE TITLE
Missing `mismatched types` error if one of the types is an alias

### ISSUE THREAD

**Issue by mchernyavsky** [intellij-rust/intellij-rust#3125]
## Environment

* **Intellij-Rust plugin version:** 0.2.0.2492-183-nightly
* **Rust toolchain version:** 1.32.0-nightly
* **IDE name and version:** IC-183.4284.148
* **Operating system:** macOS Mojave 10.14.1

## Problem description

The editor does not report the `mismatched types` error if one of the types is a type alias.

```rust
struct S;

trait T {
    type Item;
    fn foo() -> Self::Item;
}

impl T for S {
    type Item = &'static str;
    fn foo() -> Self::Item {
        let x: Self::Item = 42;
        42
    }
}
```

I expect to see the `mismatched types` error in `foo` implementation.

## Steps to reproduce

* Copy-paste code from problem description.

**Comment by mchernyavsky** [comment]
Relates to https://github.com/intellij-rust/intellij-rust/issues/3029.
